### ISSUE TITLE
Nan loss

### ISSUE THREAD

**Issue by zshyang** [facebookresearch/dino#58]
I am using the following dataset. It is a subset of ImageNet. 
```https://github.com/rmccorm4/Tiny-Imagenet-200```

After training one step, the loss just becomes nan and stops training. Have you experienced this problem? And how do you solve it?

Best regards!

**Comment by mathildecaron31** [comment]
Hi @zshyang 

Can you provide more details on your setting ?

I recommend to read carefully the documentation (`python main_dino.py --help`) since I've tried to include tips for stability. For example, the first stuff that I would try would be to try training with full precision (and not mixed precision): `--use_fp16 false`

**Comment by zshyang** [comment]
> Hi @zshyang
> 
> Can you provide more details on your setting ?
> 
> I recommend to read carefully the documentation (`python main_dino.py --help`) since I've tried to include tips for stability. For example, the first stuff that I would try would be to try training with full precision (and not mixed precision): `--use_fp16 false`

Thanks for your reply. The problem is solved with setting use_fp16 to false.

**Comment by HareshKarnan** [comment]
I get the same error with full precision training :/

**Comment by shaibagon** [comment]
I'm training on non RGB images, so my settings are different. 
I am getting NaNs for some configurations. I would like to train using fp16 (mem issues).
Is there anything else that can be done to avoid NaNs? e.g., grad clipping?

Thanks,

**Comment by kasteric** [comment]
may I ask u for the setting of training (such as the backbone and crop number) tiny Imagnet on dino? since i also encounter NaN losses after turning off fp16..

**Comment by shaibagon** [comment]
@kasteric I am using ViT base backbone, but I am not training on RGB images, so I don't think we can compare.

**Comment by HollrayChan** [comment]
I was using FP16 to train a relatively large batch of data. When training for half of the epochs, such an error occurred. I tried to add a minimum value to avoid this problem.
`loss = torch.sum((-q + 1e -8) * F.log_softmax(student_out[v] + 1e-8, dim=-1), dim=-1)`
